### PR TITLE
[SPARK-40796][BUILD][FOLLOW-UP] Fix Mypy check on unused "type: ignore"

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

The mypy check in master seems to be broken:

Screen Shot 2022-10-17 at 1 09 13 PM

(see it on https://github.com/apache/spark/pull/38279 and https://github.com/apache/spark/pull/38275, also reproducible locally).

This PR propose to remove the relevant `type: ignore` comments.

### Why are the changes needed?

Fix python lint check.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

UT

### CODE REVIEW

**Comment by amaliujia** [comment]
@HyukjinKwon @zhengruifeng

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -322,15 +322,11 @@ def _convert_measure(
...
         if type(exp) is str:
-            measure.function.arguments.append(  # type: ignore[attr-defined]
-                self.unresolved_attr(exp)
-            )
+            measure.function.arguments.append(self.unresolved_attr(exp))
         else:
-            measure.function.arguments.append(  # type: ignore[attr-defined]
-                cast(Expression, exp).to_plan(session)
-            )
+            measure.function.arguments.append(cast(Expression, exp).to_plan(session))
```
Comment (amaliujia) [inline python/pyspark/sql/connect/plan.py:329]: This is formatted by the python formatter.

**Comment by amaliujia** [comment]
Screen Shot 2022-10-17 at 3 33 02 PM

Interesting. The ignore comment was actually useful. I am confused now on how to fix that.

**Comment by amaliujia** [comment]
ah I see it is now:

The python code needs to be updated because of the proto change.

**Comment by amaliujia** [comment]
Yes. This issue is fixed. Waiting for tests passing.

**Comment by zhengruifeng** [comment]
it seems that the CI in master works well? https://github.com/apache/spark/actions/runs/3264890886/jobs/5368154706

**Comment by amaliujia** [comment]
@zhengruifeng 

I don't know how this is triggered. If you check this example PR: https://github.com/apache/spark/pull/38275. After rebasing it started to fail on this issue. That PR only touches two lines in `python/pyspark/sql/connect/plan.py` which are not relevant to any problematic lines reported by mypy.


Right now I can only suspect that it is triggered when a PR touches  `python/pyspark/sql/connect/plan.py`. Seems that all my four affected PR are in such case.

**Comment by amaliujia** [comment]
The python code on Aggregate is stale anyway: they are calling non-existing fields already.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Review by HyukjinKwon — APPROVED** [review]

**Pull request closed without merging**